src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1535,8 +1535,17 @@ protected override void ProcessRecord()
                                     try
                                     {
                                         reader = new StreamReader(StreamHelper.GetResponseStream(response));
-                                        // remove HTML tags making it easier to read
-                                        detailMsg = System.Text.RegularExpressions.Regex.Replace(reader.ReadToEnd(), "<[^>]*>", string.Empty);
+                                        if (ContentHelper.IsXml(contentType))
+                                        {
```

Comment:
I'd read all here with ReadToEnd() and pass the string to new FormatErrorMessage() where we could process XML, Json, HTML and Text.